src/driver.rs

```diff
@@ -496,6 +577,47 @@ mod tests {
         assert!(DriverManager::get_driver(0).is_ok());
     }
 
+    #[test]
+    fn test_driver_extension() {
+        // convert the driver into short_name for testing purposes
+        let drivers = |filename, is_vector| {
+            DriverManager::guess_drivers_for_write(filename, is_vector)
+                .map(|d| d.short_name())
+                .collect::<HashSet<String>>()
+        };
+        let gdal_version: i64 = crate::version::version_info("VERSION_NUM").parse().unwrap();
+        if DriverManager::get_driver_by_name("ESRI Shapefile").is_ok() {
+            assert!(drivers("test.shp", true).contains("ESRI Shapefile"));
+            assert!(drivers("my.test.shp", true).contains("ESRI Shapefile"));
+            // `shp.zip` only supported from gdal version 3.1
+            // https://gdal.org/drivers/vector/shapefile.html#compressed-files
+            if gdal_version >= 3010000 {
```

Comment:
Let's do a `if cfg!(all(major_ge_3, minor_ge_1))`, it's a little more easy to search for.

Reply:
Done.